### PR TITLE
new recipe: DataEcon

### PR DESCRIPTION
**Pull request by bbejanov**

### CODE REVIEW

File: D/DataEcon/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+sources = [
+    GitSource("https://github.com/bankofcanada/DataEcon.git",
+        "c99b6b8ea8a5d500ba8d091b384d58a815b7db01")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/DataEcon/
+# make test
+# make clean
```
Comment (giordano) [inline D/DataEcon/build_tarballs.jl:18]:
```suggestion
```

File: D/DataEcon/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+make all
```
Comment (giordano) [inline D/DataEcon/build_tarballs.jl:19]:
```suggestion
make -j${nproc} all
```

File: D/DataEcon/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+cp -p bin/* src/daec.h LICENSE.md README.md $prefix/
```
Comment (giordano) [inline D/DataEcon/build_tarballs.jl:20]:
```suggestion
install_license LICENSE.md
install -Dvm 755 "bin/sqlite3${exeext}" "${bindir}/sqlite3${exeext}"
install -Dvm 755 "bin/libdaec.${dlext}" "${libdir}/libdaec.${dlext}"
install -Dvm 644 src/daec.h "${includedir}/daec.h"
```

File: D/DataEcon/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    Platform("armv6l", "linux"; call_abi="eabihf", libc="glibc"),
+    Platform("armv7l", "linux"; call_abi="eabihf", libc="glibc"),
+    Platform("powerpc64le", "linux"; libc="glibc"),
+    Platform("i686", "linux"; libc="musl"),
+    Platform("x86_64", "linux"; libc="musl"),
+    Platform("aarch64", "linux"; libc="musl"),
+    Platform("armv6l", "linux"; call_abi="eabihf", libc="musl"),
+    Platform("armv7l", "linux"; call_abi="eabihf", libc="musl"),
+    Platform("x86_64", "freebsd";),
+]
```
Comment (giordano) [inline D/DataEcon/build_tarballs.jl:41]:
```suggestion
platforms = supported_platforms()
```

File: D/DataEcon/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct(["libdaec", "daec"], :libdaec, ["."]),
+    ExecutableProduct("sqlite3", :sqlite3shell, "."),
+    FileProduct("daec.h", :daec_header),
```
Comment (giordano) [inline D/DataEcon/build_tarballs.jl:48]:
```suggestion
    LibraryProduct(["libdaec", "daec"], :libdaec),
    ExecutableProduct("sqlite3", :sqlite3shell),
    FileProduct("include/daec.h", :daec_header),
```

File: D/DataEcon/build_tarballs.jl [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/bankofcanada/DataEcon.git",
+        "c99b6b8ea8a5d500ba8d091b384d58a815b7db01")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/DataEcon/
+make -j${nproc} all
```
Comment (giordano) [inline D/DataEcon/build_tarballs.jl:17]:
```suggestion
make -j${nproc} all LIBDE="bin/libdaec.${dlext}"
```

**Comment by bbejanov** [comment]
@giordano thanks for your help! Much appreciated!

**Comment by giordano** [comment]
Thanks for your contribution!

**Pull request merged**